lib/block-supports/duotone.php

```diff
@@ -519,3 +519,47 @@ static function () use ( $filter_svg, $selector ) {
 // Remove WordPress core filter to avoid rendering duplicate support elements.
 remove_filter( 'render_block', 'wp_render_duotone_support', 10, 2 );
 add_filter( 'render_block', 'gutenberg_render_duotone_support', 10, 2 );
+
+class WP_Duotone {
+	/**
+	 * An array of Duotone presets.
+	 *
+	 * @since 6.3.0
+	 * @var array
+	 */
+	static $duotone_presets = array();
+
+	/**
+	 * Registers the duotone preset.
+	 *
+	 * @param array $settings The block editor settings.
+	 *
+	 * @return array The block editor settings.
+	 */
+	public static function duotone_declarations( $declarations, $selector ) {
+		foreach ( $declarations as $index => $declaration ) {
+			if ( 'filter' === $declaration['name'] ) {
+				static::$duotone_presets[] = $declarations[ $index ]['value'];
+			}
+		}
+	}
+
+	public static function get_filter_svg( $duotone_preset ) {
+		$filters = '';
+		// Get the CSS variable for the preset.
+		$duotone_preset_css_var = WP_Theme_JSON_Gutenberg::get_preset_css_var( array( 'color', 'duotone' ), $duotone_preset['slug'] );
+		$duotone_preset_css_var_with_wrapper = 'var(' . $duotone_preset_css_var . ')';
+		// Only output the preset if it's used by a block.
+		if ( in_array( $duotone_preset_css_var_with_wrapper, WP_Duotone::$duotone_presets, true ) ) {
+			// Output the CSS for the preset.
+			// I think we should do this differently, but I'm not sure how.
+			$filters .= '<style>body{' . $duotone_preset_css_var . ':'. gutenberg_get_duotone_filter_property( $duotone_preset ) . ';}</style>';
```

Comment:
It would be nice if it could still be handled by `value_func` in `WP_Theme_JSON_Gutenberg` instead of moving it here.

Maybe we have to think a little more about where the filters/actions belong in that code? Could it be more useful in `gutenberg_add_global_styles_for_blocks`—modifying the `$metadata` passed into `get_styles_for_block`?